ios/ReactNativeAudioStreaming.m

```diff
@@ -431,19 +443,55 @@ - (void)unregisterRemoteControlEvents
    [commandCenter.pauseCommand removeTarget:self];
 }
 
++ (void) loadFromURL: (NSURL*) url callback:(void (^)(UIImage *image))callback {
+    dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0ul);
+    dispatch_async(queue, ^{
+        NSData * imageData = [NSData dataWithContentsOfURL:url];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            UIImage *image = [UIImage imageWithData:imageData];
+            callback(image);
+        });
+    });
+}
+
 - (void)setNowPlayingInfo:(bool)isPlaying
 {
    if (self.showNowPlayingInfo) {
-      // TODO Get artwork from stream
-      // MPMediaItemArtwork *artwork = [[MPMediaItemArtwork alloc]initWithImage:[UIImage imageNamed:@"webradio1"]];
-   
-      NSString* appName = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleName"];
-      NSDictionary *nowPlayingInfo = [NSDictionary dictionaryWithObjectsAndKeys:
-                                      self.currentSong ? self.currentSong : @"", MPMediaItemPropertyAlbumTitle,
-                                      @"", MPMediaItemPropertyAlbumArtist,
-                                      appName ? appName : @"AppName", MPMediaItemPropertyTitle,
-                                      [NSNumber numberWithFloat:isPlaying ? 1.0f : 0.0], MPNowPlayingInfoPropertyPlaybackRate, nil];
-      [MPNowPlayingInfoCenter defaultCenter].nowPlayingInfo = nowPlayingInfo;
+
+      NSString* bundleDisplayName =[[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleDisplayName"];
+      NSString* appName = bundleDisplayName? bundleDisplayName :
+         [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleName"];
+
+      NSString* title = self.currentSong ? self.currentSong : @"";
+      NSString* artist = self.currentArtist ? self.currentArtist : @"";
+      NSString* applicationName = appName ? appName : @"AppName";
+
+      if (!self.fetchedAlbumArt && self.currentArtwork != nil) {
+        NSString* imageUrl = self.currentArtwork;
+        dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+        dispatch_async(queue, ^{
+          MPNowPlayingInfoCenter *infoCenter = [MPNowPlayingInfoCenter defaultCenter];
+          UIImage *artworkImage = [UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:imageUrl]]];
+          if(artworkImage) {
+            MPMediaItemArtwork *albumArt = [[MPMediaItemArtwork alloc] initWithImage: artworkImage];
+            NSMutableDictionary *songInfo =
+              infoCenter.nowPlayingInfo ? [infoCenter.nowPlayingInfo mutableCopy] : [NSMutableDictionary dictionary];
+            [songInfo setValue:albumArt forKey:MPMediaItemPropertyArtwork];
+            infoCenter.nowPlayingInfo = songInfo;
+          }
+         self.fetchedAlbumArt = true;
```

Comment:
It seems that with your implementation you can only fetch the first song artwork, what happens if a new song from an other album is playing ?